ydb/_topic_writer/topic_writer_asyncio.py

```diff
@@ -164,65 +162,81 @@ class WriterAsyncIOReconnector:
     _update_token_interval: int
     _token_get_function: TokenGetterFuncType
     _init_message: StreamWriteMessage.InitRequest
-    _new_messages: asyncio.Queue
     _init_info: asyncio.Future
     _stream_connected: asyncio.Event
     _settings: WriterSettings
 
-    _lock: asyncio.Lock
     _last_known_seq_no: int
     _messages: Deque[InternalMessage]
     _messages_future: Deque[asyncio.Future]
-    _stop_reason: Optional[Exception]
+    _new_messages: asyncio.Queue
+    _stop_reason: asyncio.Future
     _background_tasks: List[asyncio.Task]
 
     def __init__(self, driver: SupportedDriverType, settings: WriterSettings):
         self._driver = driver
         self._credentials = driver._credentials
         self._init_message = settings.create_init_request()
-        self._new_messages = asyncio.Queue()
         self._init_info = asyncio.Future()
         self._stream_connected = asyncio.Event()
         self._settings = settings
 
-        self._lock = asyncio.Lock()
         self._last_known_seq_no = 0
         self._messages = deque()
         self._messages_future = deque()
-        self._stop_reason = None
+        self._new_messages = asyncio.Queue()
+        self._stop_reason = asyncio.Future()
         self._background_tasks = [
             asyncio.create_task(self._connection_loop(), name="connection_loop")
         ]
 
     async def close(self):
-        await self._check_stop()
-        await self._stop(TopicWriterStopped())
+        self._check_stop()
```

Comment:
do we need to raise on double close here? python libs often allows double closes (files, psycopg2. for example https://docs.python.org/3.9/library/io.html#io.IOBase.close)

Reply:
fixed